gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -420,40 +418,39 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
   /* 3. Call the filter-subplugin callback, "invoke" */
   GST_TF_FW_INVOKE_COMPAT (priv, ret, in_tensors, out_tensors);
 
-  /** @todo define enum to indicate status code */
-
-  if (ret < 0) {
-    ml_loge ("Tensor-filter invoke failed (error code = %d).\n", ret);
-    return GST_FLOW_ERROR;
-  }
-
   /* 4. Update result and free map info. */
   for (i = 0; i < prop->output_meta.num_tensors; i++) {
     if (allocate_in_invoke) {
-      GPtrArray *data_array = g_ptr_array_new ();
-      g_ptr_array_add (data_array, (gpointer) self);
-      g_ptr_array_add (data_array, (gpointer) out_tensors[i].data);
-
-      /* filter-subplugin allocated new memory, update this */
-      out_mem[i] =
-          gst_memory_new_wrapped (0, out_tensors[i].data, out_tensors[i].size,
-          0, out_tensors[i].size, (gpointer) data_array,
-          gst_tensor_filter_destroy_notify);
+      /* prepare memory block if successfully done */
+      if (ret == 0) {
+        GPtrArray *data_array = g_ptr_array_new ();
+        g_ptr_array_add (data_array, (gpointer) self);
+        g_ptr_array_add (data_array, (gpointer) out_tensors[i].data);
+
+        /* filter-subplugin allocated new memory, update this */
+        out_mem[i] = gst_memory_new_wrapped (0, out_tensors[i].data,
+            out_tensors[i].size, 0, out_tensors[i].size, (gpointer) data_array,
+            gst_tensor_filter_destroy_notify);
+      }
```

Comment:
Q. Is this fine to not handle the case `allocation_in_invoke && ret != 0`?

Reply:
ret < 0 means invoke-failure, so it is unnecessary to wrap the output data.
if ret > 0, tensor-filter will drop the incoming buffer.

Reply:
Thanks for the explanation :)